## Análise das Propostas - Crédito

<img src="images/proposta.jpg" style="width:8in;height:4in">

O BanVic possui atualmente 1996 propostas, em 4 status diferentes.
Nessa análise, vamos entender mais sobre elas.

In [1]:
# Importando os pacotes necessarios
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline
# Tamanho dos gráficos
plt.rcParams["figure.figsize"] = [10,5]
# Tamanho dos dados
plt.rcParams["font.size"] = 12

# Lendo a guia 'proposta_credito_corr' da planilha 'projeto3.xlsx'
df_propostas_credito = pd.read_excel("source/base_excel.xlsx", sheet_name = "proposta_credito_corr")

In [2]:
df_propostas_credito.head()

cod_proposta  cod_cliente  cod_colaborador data_entrada_proposta  \
0           116          338                1            2014-05-30   
1           715           45                1            2021-06-17   
2           755          494                1            2021-08-10   
3           953          381                1            2011-02-23   
4          1046          176                1            2010-11-15   

  horario_entrada_proposta  taxa_juros_mensal  valor_proposta  \
0                 18:43:12             0.0194        36199.95   
1                 13:09:32             0.0131        12897.54   
2                 04:24:02             0.0203       122236.31   
3                 07:57:58             0.0167        47784.65   
4                 16:03:28             0.0211       178393.04   

   valor_financiamento  valor_entrada  valor_prestacao  quantidade_parcelas  \
0             50032.03       13832.08          1045.22                   58   
1             19848.55        6951.01           232.12                  100   
2            187154.36       64918.05          3673.52                   56   
3             59530.61       11745.96           955.04                  109   
4            214294.19       35901.15         10219.66                   22   

   carencia status_proposta  
0         0         Enviada  
1         1         Enviada  
2         0         Enviada  
3         3         Enviada  
4         4         Enviada

Na tabela acima, temos uma visão das primeiras 5 linhas dos nosso dados referentes as propostas. 

In [3]:
df_propostas_credito.describe()

cod_proposta  cod_cliente  cod_colaborador  taxa_juros_mensal  \
count   1996.000000  1996.000000      1996.000000        1996.000000   
mean    1000.047595   505.593186        51.627255           0.016685   
std      577.196193   288.038061        28.608906           0.004937   
min        1.000000     2.000000         1.000000           0.008000   
25%      500.750000   251.750000        27.000000           0.012400   
50%      999.500000   507.000000        52.000000           0.016800   
75%     1499.250000   756.250000        76.000000           0.021000   
max     2000.000000   999.000000       100.000000           0.025000   

       valor_proposta  valor_financiamento  valor_entrada  valor_prestacao  \
count     1996.000000          1996.000000    1996.000000      1996.000000   
mean     83911.891207        124751.122665   40839.231458      4886.787114   
std      51216.157273         72780.827711   28032.143231     12897.121297   
min        944.010000          1605.630000     288.500000        19.160000   
25%      39217.425000         60521.382500   17466.550000      1196.330000   
50%      82008.300000        126780.690000   36929.670000      2328.465000   
75%     123534.172500        187196.995000   59088.237500      4019.697500   
max     206514.640000        249974.500000  120647.800000    190122.090000   

       quantidade_parcelas     carencia  
count          1996.000000  1996.000000  
mean             61.221443     3.045090  
std              35.021799     2.016466  
min               1.000000     0.000000  
25%              30.750000     1.000000  
50%              61.000000     3.000000  
75%              92.000000     5.000000  
max             120.000000     6.000000

In [4]:
df_propostas_credito.groupby('status_proposta')['cod_proposta'].describe()

count         mean         std  min     25%     50%  \
status_proposta                                                             
Aprovada              513.0   986.372320  561.350356  2.0  503.00   992.0   
Em análise            468.0   994.072650  570.618488  8.0  487.50  1000.5   
Enviada               525.0  1021.310476  595.158857  4.0  509.00  1045.0   
Validação documentos  490.0   997.289796  581.386434  1.0  503.25   986.0   

                          75%     max  
status_proposta                        
Aprovada              1453.00  2000.0  
Em análise            1477.75  1995.0  
Enviada               1563.00  1994.0  
Validação documentos  1516.75  1998.0

Pela tabela acima podemos observar que:

**Aprovada:** 513 propostas

**Em análise:** 468 propostas

**Enviada:** 525 propostas

**Validação documentos:** 490 propostas

Após analisar a base pelo SQL, podemos ver que a única variável com possibilidade de influenciar no status de uma proposta é a variável: **valor_prestacao**.


Ver arquivo: _analise_propostas.sql_

In [5]:
df_propostas_credito.groupby('status_proposta')['valor_prestacao'].describe()

count         mean           std    min       25%  \
status_proposta                                                           
Aprovada              513.0  5632.532222  14456.781525  22.34  1141.480   
Em análise            468.0  5106.121453  14574.576176  37.15  1147.255   
Enviada               525.0  4802.440019  12060.981021  19.16  1298.520   
Validação documentos  490.0  3986.922857   9979.533528  30.95  1124.050   

                           50%        75%        max  
status_proposta                                       
Aprovada              2386.290  4480.5800  156649.98  
Em análise            2265.405  3801.0875  152299.10  
Enviada               2369.390  4051.1700  190122.09  
Validação documentos  2207.360  3703.9600  136287.40

Filtrando os dados, temos a tabela abaixo:

Status                | Média | Mínimo | Máximo
--------------------- | ----- | ------ | ------     
Aprovada			  |5.632,53|	 22,34 | 156.649,98
Em análise		      |5.106,12|	 37,15 | 152.299,10
Validação documentos  |3.986,92|	 30,95 | 136.287,40
Enviada				  |4.802,44|  19,16 | 190.122,09



#### Conclusão

* Podemos dizer que propostas **menores que 22,34** ou **maiores que 156.649,98** tendem a ficar presas no status: **Enviada.**
* Temos 860 clientes com pelo menos 1 proposta de crédito.